analyzers/tests/SonarAnalyzer.Test/TestCases/NamedPlaceholdersShouldBeUnique.cs

```diff
@@ -44,6 +44,9 @@ void Compliant(ILogger logger, string foo, string bar)
             // Escaped
             logger.LogInformation("Hey {{foo}} and {{foo}}", foo, foo);      // Compliant
             logger.LogInformation("Hey {foo} and {{foo}}", foo, foo);        // Compliant
+
+            // Not string literal
+            logger.LogWarning($"Hey {foo} and {foo}", foo, foo);             // Compliant
```

Comment:
Looks good. Also add a test case with an interpolated string where the bracket characters are escaped (no need to change the implementation, it's fine to be an FN, S2629 will raise for it anyway):
```cs
logger.LogInformation($"Hey {{foo}}", foo);
```

Reply:
very good idea!